src/ert/config/gen_kw_config.py

```diff
@@ -404,64 +438,44 @@ def _sample_value(
-            "TRUNCATED_NORMAL": ["MEAN", "STD", "MIN", "MAX"],
-            "TRIANGULAR": ["MIN", "MODE", "MAX"],
-            "UNIFORM": ["MIN", "MAX"],
-            "DUNIF": ["STEPS", "MIN", "MAX"],
-            "ERRF": ["MIN", "MAX", "SKEWNESS", "WIDTH"],
-            "DERRF": ["STEPS", "MIN", "MAX", "SKEWNESS", "WIDTH"],
-            "LOGUNIF": ["MIN", "MAX"],
-            "CONST": ["VALUE"],
-            "RAW": [],
-        }
+    def _parse_transfer_function_definition(
+        t: TransformFunctionDefinition,
+    ) -> TransferFunction:
+        if t.param_func_name is None and t.values is None:
+            raise ConfigValidationError(-f"Too few instructions provided in: {t}")
+
+        if (
+            t.param_func_name not in TRANS_FUNC_ARGS
+            or t.param_func_name not in PRIOR_FUNCTIONS
+        ):
+            raise ConfigValidationError(
+                f"Unknown transfer function provided: {t.param_func_name}"
+            )
 
-        if len(param_args) > 1:
-            func_name = param_args[0]
-            param_func_name = param_args[1]
+        param_names = TRANS_FUNC_ARGS[t.param_func_name]
 
-            if (
-                param_func_name not in TRANS_FUNC_ARGS
-                or param_func_name not in PRIOR_FUNCTIONS
-            ):
-                raise ConfigValidationError(
-                    f"Unknown transfer function provided: {param_func_name}"
-                )
-
-            param_names = TRANS_FUNC_ARGS[param_func_name]
+        if len(t.values) != len(param_names):
+            raise ConfigValidationError(
+                f"Incorrect number of values provided: {t.values} "
+            )
 
-            if len(param_args) - 2 != len(param_names):
+        param_floats = []
+        for p in t.values:
+            try:
+                param_floats.append(float(p))
+            except ValueError as e:
                 raise ConfigValidationError(
-                    f"Incorrect number of values provided: {param_string}"
-                )
+                    f"Unable to convert float number: {p}"
+                ) from e
 
-            param_floats = []
-            for p in param_args[2:]:
-                try:
-                    param_floats.append(float(p))
-                except ValueError as e:
-                    raise ConfigValidationError(
-                        f"Unable to convert float number: {p}"
-                    ) from e
-
-            params = dict(zip(param_names, param_floats))
-
-            return TransferFunction(
-                name=func_name,
-                transfer_function_name=param_func_name,
-                parameter_list=params,
-                calc_func=PRIOR_FUNCTIONS[param_func_name],
-            )
+        params = dict(zip(param_names, param_floats))
 
-        else:
-            raise ConfigValidationError(
-                f"Too few instructions provided in: {param_string}"
-            )
+        return TransferFunction(
```

Comment:
Perhaps rename to `TransformFunction`. Whenever I see `TransferFunction` I think of signal processing...